src/forms/CurrentExpenses.js

```diff
@@ -41,203 +41,323 @@ import { getUnder13Expenses } from '../utils/cashflow';
+      ) : (
+        null
+      ) }
+
+      { current.hasSnap ? (
+        <ChildSupport
+          snippets    = { snippets }
+          type        = { type }
+          sharedProps = { sharedProps } />
+      ) : (
+        null
+      ) }
 
-};  // End EarnedFrom
+      {/* Head or spouse can't be a dependent, so they don't count. */}
+      { over12.length > 0 ? (
+        <DependentsOver12
+          snippets    = { snippets }
+          type        = { type }
+          sharedProps = { sharedProps } />
+      ) : (
+        null
+      ) }
 
+      { elderlyOrDisabled.length > 0 ? (
+        <ElderlyOrDisabled1
+          snippets          = { snippets }
+          type              = { type }
+          sharedProps       = { sharedProps }
+          current           = { current }
+          updateClientValue = { updateClientValue } />
+      ) : (
+        null
+      ) }
 
-const Utilities = function ({ current, type, time, updateClientValue }) {
+      {/* This comment may or may not belong here. Deeper discussion needed:
+        * These medical expenses count for Section 8 too if the disabled
+        *     person is the head or spouse. From Appendix B, item (D)
+        *     {@link http://www.tacinc.org/media/58886/S8MS%20Full%20Book.pdf} */}
+      { elderlyOrDisabledHeadOrSpouse.length > 0 || (current.hasSnap && elderlyOrDisabled.length > 0) ? (
+        <ElderlyOrDisabled2
+          snippets    = { snippets }
+          type        = { type }
+          sharedProps = { sharedProps } />
+      ) : (
+        null
+      ) }
 
-  let hasClimate     = current.climateControl,
-      hasElectricity = current.nonHeatElectricity,
-      hasPhone       = current.phone,
-      hasFuelAssist  = current.fuelAssistance;
+      <Housing
+        current={ current }
+        time={ time }
+        type={ type }
+        updateClientValue = { updateClientValue } />
 
-  let setChecked = function (evnt, inputProps) {
-    var obj = { ...inputProps, value: inputProps.checked };
-    updateClientValue(evnt, obj);
-  };  // End setChecked()
+      {/* Premature feature temporarily hidden to avoid messy revert
+        <ShowOnYes
+          clientPartial = { current }
+          propName = { `wantsToSeeOtherExpenses` }
+          updateClientValue = { updateClientValue }
+          question = { `Do you want to enter your other expenses so you can see if you need to make a different plan?` }
+          heading = { `Other Expenses` }>
+          <ExpensesOther { ...sharedProps } />
+        </ShowOnYes>
+      */}
+    </div>
+  );
 
-  // For keyboard access (already does spacebar)
-  let onKeyDown = function (evnt) {
-    if (evnt.key === `Enter`) {
-      evnt.target.click();
-    }
-  };
+};  // Ends <ExpensesFormContent>
 
-  // May want to change name to 'utilities' and value to what's 'name' now
-  // Will require more work in the change handler
+
+const Under13 = function ({ snippets, type, sharedProps, current, updateClientValue }) {
```

Comment:
I think `sharedProps` is a bit too vague; I think it would be be better to have the component actually list the props that it needs to use. The consumer of the component can still use a shared props object by calling `<Under13 {...sharedProps} />`, but the component itself should specify the minimal set of information it needs (especially if we add propTypes in the future).

This goes for the other components in the file, too.